examples/tv-casting-app/tv-casting-common/core/CastingPlayer.cpp

```diff
@@ -243,6 +253,27 @@ CHIP_ERROR CastingPlayer::StopConnecting()
     mTargetCastingPlayer.reset();
     CastingPlayerDiscovery::GetInstance()->ClearCastingPlayersInternal();
 
+    // shouldSendIdentificationDeclarationMessage is false only when StopConnecting() is called by the casting library.
+    if (!shouldSendIdentificationDeclarationMessage)
+    {
+        ChipLogProgress(AppServer,
+                        "CastingPlayer::StopConnecting() shouldSendIdentificationDeclarationMessage: %s, User Directed "
+                        "Commissioning stopped by CastingPlayer/Commissioner user",
+                        shouldSendIdentificationDeclarationMessage ? "true" : "false");
+        // If the client has not provided the (Optional) ConnectionCallbacks mCommissionerDeclarationCallback, we can call the
+        // ConnectionCallbacks mOnConnectionComplete callback with CHIP_ERROR_CONNECTION_ABORTED, to alert the client that the
+        // commissioning session was cancelled by the CastingPlayer/Commissioner user. For example, in the scenario where user 1 is
+        // controlling the casting Client/Commissionee and user 2 is controlling the CastingPlayer/Commissioner.
+        if (!mClientProvidedCommissionerDeclarationCallback)
+        {
+            ChipLogProgress(AppServer,
+                            "CastingPlayer::StopConnecting() CommissionerDeclarationCallback not provided by client, calling "
+                            "mOnConnectionComplete with CHIP_ERROR_CONNECTION_ABORTED");
+            mOnCompleted(CHIP_ERROR_CONNECTION_ABORTED, nullptr);
+        }
```

Comment:
I am not sure if we need this? Don't we already callback on CommissionerDeclarationCallback, if it is provided. If that callback was not provided, we can simplify and assume the client is not interested in getting the callback. I'd avoid creating this "either-or" logic with the callbacks.

Reply:
We're calling the CommissionerDeclarationCallback optional but the Client will be forced to implement it for this common use case/scenario:

In this scenario there are multiple CastingPlayers TV in the user environment. Therefore the Casting Client/Commissionee user (user 1) is not necessarily the same user who's controlling the CastingPlayer/Commissioner TV (user 2). If user 2 cancels, we should alert user 1 that connecting was aborted as soon as possible, so user 1 does not need to wait for the timeout. This is a useful user feature and reduces Client effort.